tests/test_intersection_over_union.py

```diff
@@ -13,18 +13,17 @@ def test_iou():
     
     iou = biau.label.intersection_over_union(a, b)
     
-    assert(iou.shape == (size, size))
+    assert(iou.shape == (len(a), len(b)))
```

Comment:
```suggestion
    assert(iou.shape == (np.max(a), np.max(b)))
```
Just a quick question: Would this also work and be more general? I mean with any label images and not just with `range()`-made label images?

Reply:
I *think* the generalized solution would be something like
```python
assert(iou.shape == (len(np.unique(a)), len(np.unique(b)))
```

I'll try around a bit.

Reply:
I think this would be a more generalized test:

```python
    # Create two arrays of random length
    size = np.random.randint(5, 10)
    low = 1
    a = np.arange(low, size, 1, dtype=np.uint8)
    b = np.arange(low, size, 1, dtype=np.uint8)
    
    # Shuffle array
    np.random.shuffle(b)
    
    # delete random entries from a and b
    idx = np.random.randint(low, size-1, 1)
    b[idx] = 0
    a[idx] = 0
    
    iou = biau.label.intersection_over_union_matrix(a, b)
    
    assert(iou.shape == (len(np.unique([a, b])), len(np.unique([a, b]))))
```

As I see it, `confusion_matrix` from sklearn will always return a square matrix, even if the number of unique labels is different in the two input labels. Depending on the present labels, there will be some empty columns, which should be handled by `relabel_sequential` in https://github.com/BiAPoL/biapol-utilities/pull/20#discussion_r745445866.

As in https://github.com/BiAPoL/biapol-utilities/pull/20#discussion_r745442795, this is a bit of a tautological test right now, but could become useful if we eventually decide to re-implement the confusion_matrix calculation.